### PR TITLE
Improve genezio error handling

### PR DESCRIPTION
**Pull request by andreia-oca**
# Pull Request Template

## Type of change

- [x] 🧑‍💻 Improvement

## Description

This PR improves the genezio codebase in the following ways:
- Removes the genezio init command because it's deprecated in favor of genezio super command.
- Adds types for the each command options
- Improves error handling - each method should throw errors to the caller. In `genezio.ts` all the errors are caught and displayed in the user's terminal. This PR does this for most of the "shallow" commands such as account, list, login etc. For the monsters - `deploy` and `local` - I will come back with a subsequent PR.

## Checklist

- [x] My code follows the contributor guidelines of this project;
- [x] New and existing unit tests pass locally with my changes;

### CODE REVIEW

File: src/commands/delete.ts [hunk 1]
```diff
@@ -34,18 +31,21 @@ export async function deleteCommand(projectId: string, options: any) {
...
     // show prompt if no project id is selected
     if (typeof projectId === "string" && projectId.trim().length === 0) {
         const spinner = new Spinner("%s  ");
         spinner.setSpinnerString("|/-\\");
         spinner.start();
-        const projectsJson = await listProjects();
+        const projectsJson = await listProjects().catch((error) => {
+            throw error;
+        });
+
```
Comment (costinsin) [inline src/commands/delete.ts:43]: This is uncessary. Any awaited async functions throw automatically if there is no `try catch` block around them, or have `.catch` call attached to them.

You can test it yourself here:
https://www.typescriptlang.org/play?#code/IYZwngdgxgBAZgV2gFwJYHsI2QCwE7oDuIAongXgBQCUMA3gFAzPb5EwBEApuenhwG4GAXwYNQkWIhQYsAE3QBldAFsuuVBADmNekxbBCwVMlYFiZCjRFiJ0eEihpMMFcYi7GLbHjB7v3obGpgrKahraNELewjBQwMhQODCUPHi0XgHMUJgg6AA2XAB0+eg6adTRLKKiDG6a1kA

```suggestion
        const projectsJson = await listProjects();
```

File: src/commands/delete.ts [hunk 2]
```diff
@@ -86,6 +86,9 @@ async function deleteProjectHandler(projectId: string, forced: boolean) {
         }
     }
 
-    const status = await deleteProject(projectId);
+    const status = await deleteProject(projectId).catch((error) => {
+        throw error;
+    });
+
```
Comment (costinsin) [inline src/commands/delete.ts:92]: Please revert it as it was.

File: src/commands/generateSdk.ts [hunk 3]
```diff
@@ -143,7 +130,7 @@ async function generateRemoteSdkHandler(
     region: string,
 ) {
     // get all project classes
-    const projects = await listProjects(0).catch((error: any) => {
+    const projects = await listProjects(0).catch((error) => {
         throw error;
     });
```
Comment (costinsin) [inline src/commands/generateSdk.ts:135]: Let's remove the `.catch` even if you didn't add it.

File: src/commands/generateSdk.ts [hunk 4]
```diff
@@ -153,7 +140,9 @@ async function generateRemoteSdkHandler(
     );
 
     // get project info
-    const projectEnv = await getProjectEnvFromProject(project.id, stage);
+    const projectEnv = await getProjectEnvFromProject(project.id, stage).catch((error) => {
+        throw error;
+    });
```
Comment (costinsin) [inline src/commands/generateSdk.ts:145]: Please revert it as it was.

**Review by costinsin — CHANGES_REQUESTED** [review]
I think this is a very nice change that improves the code of the CLI. Thank you for doing it!

I have left some comments related more to TypeScript coding style, than the logic of the program.

File: src/commands/delete.ts (hunk 1 above)
Comment (andreia-oca) [reply src/commands/delete.ts]: I didn't know that - thanks for the playground. I reverted the catch blocks.

**Review by costinsin — APPROVED** [review]
[ok](https://app.graphite.dev/macros/Genez-io)

**Pull request merged**